src/commands/moderators/activeThreads.ts

```diff
@@ -97,8 +89,75 @@ export default {
   },
 } satisfies CommandHandlerConfig
 
-async function getActiveThreads(client: Client, guild: { id: string }) {
-  return (await client.rest.get(
-    Routes.guildActiveThreads(guild.id),
-  )) as RESTGetAPIGuildThreadsResult
+async function generateReport(
+  botContext: BotContext,
+  interaction: MessageComponentInteraction,
+  threads: APIThreadChannel[],
+) {
+  const tsv = generateTsv([
+    [
+      'ID',
+      'Name',
+      'Channel ID',
+      'Channel Name',
+      'Last Message',
+      'Message Count',
+    ],
+    ...threads.map((thread) => {
+      const channel = thread.parent_id
+        ? botContext.client.channels.cache.get(thread.parent_id)
+        : undefined
+      return [
+        thread.id,
+        thread.name,
+        thread.parent_id ?? '-',
+        (channel && 'name' in channel && channel.name) ?? '-',
+        thread.last_message_id
+          ? toLocalDate(SnowflakeUtil.timestampFrom(thread.last_message_id))
+          : '-',
+        thread.message_count,
+      ]
+    }),
+  ])
+  await interaction.reply({
+    content: 'Here is the report.',
+    files: [new AttachmentBuilder(Buffer.from(tsv), { name: 'threads.tsv' })],
+    ephemeral: true,
+  })
+}
+
+async function pruneThreads(
+  botContext: BotContext,
+  selectInteraction: MessageComponentInteraction,
+  threadIds: string[],
+) {
+  await selectInteraction.reply({
+    content: `Archiving ${threadIds.length} threads...`,
+    ephemeral: true,
+  })
+
+  try {
+    let lastUpdate = 0
+    let count = 0
+    for (const threadId of threadIds) {
+      const thread = botContext.client.channels.cache.get(threadId)
+      if (!thread || !('setArchived' in thread)) continue
+      await thread.setArchived(true)
+      count++
+      if (Date.now() - lastUpdate > 5e3) {
```

Comment:
This is for debouncing the pruning command in 5 seconds interval right?

Reply:
no, it’s for throttling krub